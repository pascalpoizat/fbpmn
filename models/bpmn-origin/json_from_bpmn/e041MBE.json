{
    "isInterrupt": {
        "NonInterruptBE": false,
        "BoundaryEvent_0ofx7yq": false,
        "InterruptBE": true,
        "BoundaryEvent_0lkn8pc": true
    },
    "catN": {
        "StartEvent_1s47y70": "NoneStartEvent",
        "NonInterruptBE": "MessageBoundaryEvent",
        "StartEvent_1": "NoneStartEvent",
        "EndEvent_0rzkmq3": "NoneEndEvent",
        "ExclusiveGateway_0xuurqz": "XorGateway",
        "ExclusiveGateway_1dkr7pq": "XorGateway",
        "SInterrupt": "SendTask",
        "Qid": "Process",
        "StartEvent_1wbfgbp": "NoneStartEvent",
        "SubProcess_0gsfmc3": "SubProcess",
        "Pid": "Process",
        "TA": "AbstractTask",
        "SInfo": "SendTask",
        "BoundaryEvent_0ofx7yq": "TimerBoundaryEvent",
        "EndEvent_009ffjw": "NoneEndEvent",
        "InterruptBE": "MessageBoundaryEvent",
        "EndEvent_08qbw2d": "TerminateEndEvent",
        "EndEvent_086sfn7": "NoneEndEvent",
        "EndEvent_14tvnqu": "NoneEndEvent",
        "BoundaryEvent_0lkn8pc": "TimerBoundaryEvent"
    },
    "catE": {
        "SequenceFlow_1sctg40": "NormalSequenceFlow",
        "SequenceFlow_0v3rr2b": "NormalSequenceFlow",
        "SequenceFlow_1qzgvib": "NormalSequenceFlow",
        "SequenceFlow_1wjaklx": "NormalSequenceFlow",
        "SequenceFlow_1tf20c8": "NormalSequenceFlow",
        "SequenceFlow_0zcc527": "NormalSequenceFlow",
        "SequenceFlow_1hkg88d": "NormalSequenceFlow",
        "SequenceFlow_19yqu7y": "NormalSequenceFlow",
        "SequenceFlow_0mtfqyq": "NormalSequenceFlow",
        "SequenceFlow_0h86znt": "NormalSequenceFlow",
        "SequenceFlow_13lhzc9": "NormalSequenceFlow",
        "SequenceFlow_1c63kjk": "NormalSequenceFlow",
        "MessageFlow_12lpwl8": "MessageFlow",
        "SequenceFlow_0c9btso": "NormalSequenceFlow",
        "SequenceFlow_1uh9c90": "NormalSequenceFlow",
        "MessageFlow_0vpnljc": "MessageFlow"
    },
    "containN": {
        "Qid": [
            "StartEvent_1s47y70",
            "ExclusiveGateway_0xuurqz",
            "EndEvent_14tvnqu",
            "SInfo",
            "ExclusiveGateway_1dkr7pq",
            "SInterrupt"
        ],
        "SubProcess_0gsfmc3": [
            "StartEvent_1wbfgbp",
            "TA",
            "EndEvent_009ffjw"
        ],
        "Pid": [
            "SubProcess_0gsfmc3",
            "InterruptBE",
            "StartEvent_1",
            "EndEvent_0rzkmq3",
            "BoundaryEvent_0ofx7yq",
            "NonInterruptBE",
            "BoundaryEvent_0lkn8pc",
            "EndEvent_08qbw2d",
            "EndEvent_086sfn7"
        ]
    },
    "messageE": {
        "MessageFlow_12lpwl8": "info",
        "MessageFlow_0vpnljc": "interrupt"
    },
    "timeInformation": {
        "BoundaryEvent_0ofx7yq": {
            "timerDefinitionValue": "P10M",
            "timerDefinitionType": "TimeDuration"
        },
        "BoundaryEvent_0lkn8pc": {
            "timerDefinitionValue": "P30M",
            "timerDefinitionType": "TimeDuration"
        }
    },
    "targetE": {
        "SequenceFlow_1sctg40": "SInterrupt",
        "SequenceFlow_0v3rr2b": "EndEvent_14tvnqu",
        "SequenceFlow_1qzgvib": "EndEvent_086sfn7",
        "SequenceFlow_1wjaklx": "EndEvent_0rzkmq3",
        "SequenceFlow_1tf20c8": "SInfo",
        "SequenceFlow_0zcc527": "TA",
        "SequenceFlow_1hkg88d": "SubProcess_0gsfmc3",
        "SequenceFlow_19yqu7y": "ExclusiveGateway_0xuurqz",
        "SequenceFlow_0mtfqyq": "EndEvent_08qbw2d",
        "SequenceFlow_0h86znt": "EndEvent_009ffjw",
        "SequenceFlow_13lhzc9": "EndEvent_086sfn7",
        "SequenceFlow_1c63kjk": "ExclusiveGateway_1dkr7pq",
        "MessageFlow_12lpwl8": "NonInterruptBE",
        "SequenceFlow_0c9btso": "ExclusiveGateway_1dkr7pq",
        "SequenceFlow_1uh9c90": "EndEvent_08qbw2d",
        "MessageFlow_0vpnljc": "InterruptBE"
    },
    "name": "e041MBE",
    "attached": {
        "NonInterruptBE": "SubProcess_0gsfmc3",
        "BoundaryEvent_0ofx7yq": "SubProcess_0gsfmc3",
        "InterruptBE": "SubProcess_0gsfmc3",
        "BoundaryEvent_0lkn8pc": "SubProcess_0gsfmc3"
    },
    "containE": {
        "Qid": [
            "SequenceFlow_19yqu7y",
            "SequenceFlow_1sctg40",
            "SequenceFlow_1tf20c8",
            "SequenceFlow_1c63kjk",
            "SequenceFlow_0c9btso",
            "SequenceFlow_0v3rr2b"
        ],
        "SubProcess_0gsfmc3": [
            "SequenceFlow_0zcc527",
            "SequenceFlow_0h86znt"
        ],
        "Pid": [
            "SequenceFlow_1hkg88d",
            "SequenceFlow_1wjaklx",
            "SequenceFlow_0mtfqyq",
            "SequenceFlow_1qzgvib",
            "SequenceFlow_1uh9c90",
            "SequenceFlow_13lhzc9"
        ]
    },
    "edges": [
        "MessageFlow_0vpnljc",
        "MessageFlow_12lpwl8",
        "SequenceFlow_1hkg88d",
        "SequenceFlow_1wjaklx",
        "SequenceFlow_0mtfqyq",
        "SequenceFlow_1qzgvib",
        "SequenceFlow_1uh9c90",
        "SequenceFlow_13lhzc9",
        "SequenceFlow_0zcc527",
        "SequenceFlow_0h86znt",
        "SequenceFlow_19yqu7y",
        "SequenceFlow_1sctg40",
        "SequenceFlow_1tf20c8",
        "SequenceFlow_1c63kjk",
        "SequenceFlow_0c9btso",
        "SequenceFlow_0v3rr2b"
    ],
    "messages": [
        "interrupt",
        "info"
    ],
    "sourceE": {
        "SequenceFlow_1sctg40": "ExclusiveGateway_0xuurqz",
        "SequenceFlow_0v3rr2b": "ExclusiveGateway_1dkr7pq",
        "SequenceFlow_1qzgvib": "InterruptBE",
        "SequenceFlow_1wjaklx": "SubProcess_0gsfmc3",
        "SequenceFlow_1tf20c8": "ExclusiveGateway_0xuurqz",
        "SequenceFlow_0zcc527": "StartEvent_1wbfgbp",
        "SequenceFlow_1hkg88d": "StartEvent_1",
        "SequenceFlow_19yqu7y": "StartEvent_1s47y70",
        "SequenceFlow_0mtfqyq": "NonInterruptBE",
        "SequenceFlow_0h86znt": "TA",
        "SequenceFlow_13lhzc9": "BoundaryEvent_0lkn8pc",
        "SequenceFlow_1c63kjk": "SInterrupt",
        "MessageFlow_12lpwl8": "SInfo",
        "SequenceFlow_0c9btso": "SInfo",
        "SequenceFlow_1uh9c90": "BoundaryEvent_0ofx7yq",
        "MessageFlow_0vpnljc": "SInterrupt"
    },
    "nodes": [
        "Pid",
        "Qid",
        "SubProcess_0gsfmc3",
        "InterruptBE",
        "StartEvent_1",
        "EndEvent_0rzkmq3",
        "BoundaryEvent_0ofx7yq",
        "NonInterruptBE",
        "BoundaryEvent_0lkn8pc",
        "EndEvent_08qbw2d",
        "EndEvent_086sfn7",
        "StartEvent_1wbfgbp",
        "TA",
        "EndEvent_009ffjw",
        "StartEvent_1s47y70",
        "ExclusiveGateway_0xuurqz",
        "EndEvent_14tvnqu",
        "SInfo",
        "ExclusiveGateway_1dkr7pq",
        "SInterrupt"
    ],
    "nameN": {
        "SInterrupt": "Send Interrupt",
        "TA": "TA",
        "SInfo": "Send Information",
        "BoundaryEvent_0ofx7yq": "10 minutes",
        "BoundaryEvent_0lkn8pc": "30 minutes"
    }
}