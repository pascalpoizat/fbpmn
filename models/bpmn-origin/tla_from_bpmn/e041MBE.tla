---------------- MODULE e041MBE ----------------

EXTENDS TLC, PWSTypes

VARIABLES nodemarks, edgemarks, net

Interest ==
  "Pid" :> { "interrupt", "info" }
  @@ "Qid" :> {  }

ContainRel ==
  "Pid" :> { "SubProcess_0gsfmc3", "InterruptBE", "StartEvent_1", "EndEvent_0rzkmq3", "BoundaryEvent_0ofx7yq", "NonInterruptBE", "BoundaryEvent_0lkn8pc", "EndEvent_08qbw2d", "EndEvent_086sfn7" }
  @@ "Qid" :> { "StartEvent_1s47y70", "ExclusiveGateway_0xuurqz", "EndEvent_14tvnqu", "SInfo", "ExclusiveGateway_1dkr7pq", "SInterrupt" }
  @@ "SubProcess_0gsfmc3" :> { "StartEvent_1wbfgbp", "TA", "EndEvent_009ffjw" }

Node == { "Pid", "Qid", "SubProcess_0gsfmc3", "InterruptBE", "StartEvent_1", "EndEvent_0rzkmq3", "BoundaryEvent_0ofx7yq", "NonInterruptBE", "BoundaryEvent_0lkn8pc", "EndEvent_08qbw2d", "EndEvent_086sfn7", "StartEvent_1wbfgbp", "TA", "EndEvent_009ffjw", "StartEvent_1s47y70", "ExclusiveGateway_0xuurqz", "EndEvent_14tvnqu", "SInfo", "ExclusiveGateway_1dkr7pq", "SInterrupt" }

Edge == { "MessageFlow_0vpnljc", "MessageFlow_12lpwl8", "SequenceFlow_1hkg88d", "SequenceFlow_1wjaklx", "SequenceFlow_0mtfqyq", "SequenceFlow_1qzgvib", "SequenceFlow_1uh9c90", "SequenceFlow_13lhzc9", "SequenceFlow_0zcc527", "SequenceFlow_0h86znt", "SequenceFlow_19yqu7y", "SequenceFlow_1sctg40", "SequenceFlow_1tf20c8", "SequenceFlow_1c63kjk", "SequenceFlow_0c9btso", "SequenceFlow_0v3rr2b" }

Message == { "interrupt", "info" }

msgtype ==
   "MessageFlow_0vpnljc" :> "interrupt"
@@ "MessageFlow_12lpwl8" :> "info"


source ==
   "MessageFlow_0vpnljc" :> "SInterrupt"
@@ "MessageFlow_12lpwl8" :> "SInfo"
@@ "SequenceFlow_1hkg88d" :> "StartEvent_1"
@@ "SequenceFlow_1wjaklx" :> "SubProcess_0gsfmc3"
@@ "SequenceFlow_0mtfqyq" :> "NonInterruptBE"
@@ "SequenceFlow_1qzgvib" :> "InterruptBE"
@@ "SequenceFlow_1uh9c90" :> "BoundaryEvent_0ofx7yq"
@@ "SequenceFlow_13lhzc9" :> "BoundaryEvent_0lkn8pc"
@@ "SequenceFlow_0zcc527" :> "StartEvent_1wbfgbp"
@@ "SequenceFlow_0h86znt" :> "TA"
@@ "SequenceFlow_19yqu7y" :> "StartEvent_1s47y70"
@@ "SequenceFlow_1sctg40" :> "ExclusiveGateway_0xuurqz"
@@ "SequenceFlow_1tf20c8" :> "ExclusiveGateway_0xuurqz"
@@ "SequenceFlow_1c63kjk" :> "SInterrupt"
@@ "SequenceFlow_0c9btso" :> "SInfo"
@@ "SequenceFlow_0v3rr2b" :> "ExclusiveGateway_1dkr7pq"

target ==
   "MessageFlow_0vpnljc" :> "InterruptBE"
@@ "MessageFlow_12lpwl8" :> "NonInterruptBE"
@@ "SequenceFlow_1hkg88d" :> "SubProcess_0gsfmc3"
@@ "SequenceFlow_1wjaklx" :> "EndEvent_0rzkmq3"
@@ "SequenceFlow_0mtfqyq" :> "EndEvent_08qbw2d"
@@ "SequenceFlow_1qzgvib" :> "EndEvent_086sfn7"
@@ "SequenceFlow_1uh9c90" :> "EndEvent_08qbw2d"
@@ "SequenceFlow_13lhzc9" :> "EndEvent_086sfn7"
@@ "SequenceFlow_0zcc527" :> "TA"
@@ "SequenceFlow_0h86znt" :> "EndEvent_009ffjw"
@@ "SequenceFlow_19yqu7y" :> "ExclusiveGateway_0xuurqz"
@@ "SequenceFlow_1sctg40" :> "SInterrupt"
@@ "SequenceFlow_1tf20c8" :> "SInfo"
@@ "SequenceFlow_1c63kjk" :> "ExclusiveGateway_1dkr7pq"
@@ "SequenceFlow_0c9btso" :> "ExclusiveGateway_1dkr7pq"
@@ "SequenceFlow_0v3rr2b" :> "EndEvent_14tvnqu"

CatN ==
   "Pid" :> Process
@@ "Qid" :> Process
@@ "SubProcess_0gsfmc3" :> SubProcess
@@ "InterruptBE" :> MessageBoundaryEvent
@@ "StartEvent_1" :> NoneStartEvent
@@ "EndEvent_0rzkmq3" :> NoneEndEvent
@@ "BoundaryEvent_0ofx7yq" :> TimerBoundaryEvent
@@ "NonInterruptBE" :> MessageBoundaryEvent
@@ "BoundaryEvent_0lkn8pc" :> TimerBoundaryEvent
@@ "EndEvent_08qbw2d" :> TerminateEndEvent
@@ "EndEvent_086sfn7" :> NoneEndEvent
@@ "StartEvent_1wbfgbp" :> NoneStartEvent
@@ "TA" :> AbstractTask
@@ "EndEvent_009ffjw" :> NoneEndEvent
@@ "StartEvent_1s47y70" :> NoneStartEvent
@@ "ExclusiveGateway_0xuurqz" :> ExclusiveOr
@@ "EndEvent_14tvnqu" :> NoneEndEvent
@@ "SInfo" :> SendTask
@@ "ExclusiveGateway_1dkr7pq" :> ExclusiveOr
@@ "SInterrupt" :> SendTask

CatE ==
   "MessageFlow_0vpnljc" :> MessageFlow
@@ "MessageFlow_12lpwl8" :> MessageFlow
@@ "SequenceFlow_1hkg88d" :> NormalSeqFlow
@@ "SequenceFlow_1wjaklx" :> NormalSeqFlow
@@ "SequenceFlow_0mtfqyq" :> NormalSeqFlow
@@ "SequenceFlow_1qzgvib" :> NormalSeqFlow
@@ "SequenceFlow_1uh9c90" :> NormalSeqFlow
@@ "SequenceFlow_13lhzc9" :> NormalSeqFlow
@@ "SequenceFlow_0zcc527" :> NormalSeqFlow
@@ "SequenceFlow_0h86znt" :> NormalSeqFlow
@@ "SequenceFlow_19yqu7y" :> NormalSeqFlow
@@ "SequenceFlow_1sctg40" :> NormalSeqFlow
@@ "SequenceFlow_1tf20c8" :> NormalSeqFlow
@@ "SequenceFlow_1c63kjk" :> NormalSeqFlow
@@ "SequenceFlow_0c9btso" :> NormalSeqFlow
@@ "SequenceFlow_0v3rr2b" :> NormalSeqFlow

LOCAL preEdges ==
  [ i \in {} |-> {}]
PreEdges(n,e) == preEdges[n,e]

PreNodes(n,e) == { target[ee] : ee \in preEdges[n,e] }
          \union { nn \in { source[ee] : ee \in preEdges[n,e] } : CatN[nn] \in { NoneStartEvent, MessageStartEvent } }

BoundaryEvent ==
   "InterruptBE" :> [ attachedTo |-> "SubProcess_0gsfmc3", cancelActivity |-> TRUE ]
@@ "NonInterruptBE" :> [ attachedTo |-> "SubProcess_0gsfmc3", cancelActivity |-> FALSE ]
@@ "BoundaryEvent_0ofx7yq" :> [ attachedTo |-> "SubProcess_0gsfmc3", cancelActivity |-> FALSE ]
@@ "BoundaryEvent_0lkn8pc" :> [ attachedTo |-> "SubProcess_0gsfmc3", cancelActivity |-> TRUE ]

WF == INSTANCE PWSWellFormed
ASSUME WF!WellTyped
ASSUME WF!WellFormedness

ConstraintNode == TRUE \* none
ConstraintEdge == TRUE \* none
Constraint == TRUE     \* none
INSTANCE PWSConstraints
INSTANCE UserProperties
INSTANCE PWSSemantics

================================================================

