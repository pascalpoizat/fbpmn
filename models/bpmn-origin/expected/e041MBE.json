{
  "e041MBE": {
    "Network02FifoPair": {
      "Prop02Safe": {
        "value": false,
        "states": null,
        "trans": null,
        "depth": "5"
      },
      "Prop04MsgSound": {
        "value": true,
        "states": "604",
        "trans": "2081",
        "depth": "19"
      },
      "Prop03Sound": {
        "value": false,
        "states": "604",
        "trans": "2081",
        "depth": null
      },
      "Prop01Type": {
        "value": true,
        "states": "604",
        "trans": "2081",
        "depth": "19"
      }
    },
    "Network03Causal": {
      "Prop02Safe": {
        "value": false,
        "states": null,
        "trans": null,
        "depth": "5"
      },
      "Prop04MsgSound": {
        "value": true,
        "states": "604",
        "trans": "2081",
        "depth": "19"
      },
      "Prop03Sound": {
        "value": false,
        "states": "604",
        "trans": "2081",
        "depth": null
      },
      "Prop01Type": {
        "value": true,
        "states": "604",
        "trans": "2081",
        "depth": "19"
      }
    },
    "Network07RSC": {
      "Prop02Safe": {
        "value": false,
        "states": null,
        "trans": null,
        "depth": "5"
      },
      "Prop04MsgSound": {
        "value": true,
        "states": "604",
        "trans": "2081",
        "depth": "19"
      },
      "Prop03Sound": {
        "value": false,
        "states": "604",
        "trans": "2081",
        "depth": null
      },
      "Prop01Type": {
        "value": true,
        "states": "604",
        "trans": "2081",
        "depth": "19"
      }
    },
    "Network05Outbox": {
      "Prop02Safe": {
        "value": false,
        "states": null,
        "trans": null,
        "depth": "5"
      },
      "Prop04MsgSound": {
        "value": true,
        "states": "604",
        "trans": "2081",
        "depth": "19"
      },
      "Prop03Sound": {
        "value": false,
        "states": "604",
        "trans": "2081",
        "depth": null
      },
      "Prop01Type": {
        "value": true,
        "states": "604",
        "trans": "2081",
        "depth": "19"
      }
    },
    "Network04Inbox": {
      "Prop02Safe": {
        "value": false,
        "states": null,
        "trans": null,
        "depth": "5"
      },
      "Prop04MsgSound": {
        "value": true,
        "states": "604",
        "trans": "2081",
        "depth": "19"
      },
      "Prop03Sound": {
        "value": false,
        "states": "604",
        "trans": "2081",
        "depth": null
      },
      "Prop01Type": {
        "value": true,
        "states": "604",
        "trans": "2081",
        "depth": "19"
      }
    },
    "Network01Bag": {
      "Prop02Safe": {
        "value": false,
        "states": null,
        "trans": null,
        "depth": "5"
      },
      "Prop04MsgSound": {
        "value": true,
        "states": "604",
        "trans": "2081",
        "depth": "19"
      },
      "Prop03Sound": {
        "value": false,
        "states": "604",
        "trans": "2081",
        "depth": null
      },
      "Prop01Type": {
        "value": true,
        "states": "604",
        "trans": "2081",
        "depth": "19"
      }
    },
    "Network06Fifo": {
      "Prop02Safe": {
        "value": false,
        "states": null,
        "trans": null,
        "depth": "5"
      },
      "Prop04MsgSound": {
        "value": true,
        "states": "604",
        "trans": "2081",
        "depth": "19"
      },
      "Prop03Sound": {
        "value": false,
        "states": "604",
        "trans": "2081",
        "depth": null
      },
      "Prop01Type": {
        "value": true,
        "states": "604",
        "trans": "2081",
        "depth": "19"
      }
    }
  }
}
